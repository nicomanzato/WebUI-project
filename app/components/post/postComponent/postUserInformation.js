import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableHighlight
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export const PostUserInformation = (props) => {
  return (
    <View style={styles.usernameTitle}>
      <Text style={styles.profileUsername}>{props.user.name}</Text>
      {props.user.verified === true &&
        <Ionicons name="ios-checkmark-circle" size={16} color="#1FBFFF" />
      }
      <Text style={styles.profileScreenName}> @{props.user.screen_name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({

  usernameTitle: {
    flexDirection: 'row',
  },

  profileUsername: {
    fontWeight: "600",
  },

  profileScreenName: {
    color: 'grey'
  },

});
