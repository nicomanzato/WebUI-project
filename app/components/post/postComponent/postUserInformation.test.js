import React from 'react'
import {PostUserInformation} from './postUserInformation'
import PostMock from './../../../store/post/mock/postMock'
import { shallow, mount, ReactWrapper } from 'enzyme'

describe('post component', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <PostUserInformation user={PostMock.user} />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
