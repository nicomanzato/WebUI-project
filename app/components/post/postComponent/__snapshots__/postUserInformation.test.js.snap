// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component should render correctly 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "600",
      }
    }
  >
    TN - Todo Noticias
  </Text>
  <Icon
    allowFontScaling={false}
    color="#1FBFFF"
    name="ios-checkmark-circle"
    size={16}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "grey",
      }
    }
  >
     @
    todonoticias
  </Text>
</Component>
`;
